---
title: "CC2 Rade De Brest"
output: github_document
author: Mrozinski Alexandre
token: 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
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo=TRUE, eval=FALSE)
library(phyloseq)
library(dada2)
library(DECIPHER)
library(phangorn)
library(ggplot2)
library(gridExtra)
library(shiny)
library(miniUI)
library(caret)
library(pls)
library(e1071)
library(ggplot2)
library(randomForest)
library(dplyr)
library(ggrepel)
#library(nlme)
library(devtools)
library(reshape2)
library(PMA)
#library(structSSI)
library(ade4)
library(ggnetwork)
library(intergraph)
library(scales)
library(genefilter)
library(impute)
library(phyloseqGraphTest)
library(Biostrings)
```

```{bash}
knitr::opts_chunk$set(echo=TRUE, eval=FALSE)

wget pagesperso.univ-brest.fr/~maignien/teaching/M1-MFA/UE-Ecogenomique2/EcoG2_data_cc2.tar.gz
tar xzvf EcoG2_data_cc2.tar.gz
```

```{bash}
knitr::opts_chunk$set(echo=TRUE, eval=FALSE)

mkdir data
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=FALSE)

path <- "data"
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

list.files(path)
```


```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

fnFs <- sort(list.files(path, pattern="_R1", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2", full.names = TRUE))

sample.names <- sapply(strsplit(basename(fnFs), "_R"), `[`, 1)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

plotQualityProfile(fnRs[1:2])
plotQualityProfile(fnFs[1:2])
```

#Filter and trim

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,190),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)
head(out)
```

#Learn the Error Rates

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

errFs <- learnErrors(filtFs, multithread=TRUE)
errRs <- learnErrors(filtRs, multithread=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

plotErrors(errFs, nominalQ=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

plotErrors(errRs, nominalQ=TRUE)
```

#Sample Inference

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

dadaFs <- dada(filtFs, err=errFs, multithread=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

dadaRs <- dada(filtRs, err=errRs, multithread=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

dadaFs[[1]]
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

dadaRs[[1]]
```

#Merge paired reads

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)

head(mergers[[1]])
```

#Construct sequence table
```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

table(nchar(getSequences(seqtab)))
```

#Remove chimeras
```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

sum(seqtab.nochim)/sum(seqtab)
```

#Track reads through the pipeline
```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))

colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
```

#Assign taxonomy
```{bash}
knitr::opts_chunk$set(echo=TRUE, eval=FALSE)
wget https://zenodo.org/record/4587955/files/silva_nr99_v138.1_train_set.fa.gz?download=1
```
```{bash}
knitr::opts_chunk$set(echo=FALSE, eval=FALSE)
#test taxo 2
wget https://zenodo.org/record/4587955/files/silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1
```
```{bash}
knitr::opts_chunk$set(echo=FALSE, eval=FALSE)
#test taxo 3
wget https://zenodo.org/record/4587955/files/silva_species_assignment_v138.1.fa.gz?download=1
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

taxa <- assignTaxonomy(seqtab.nochim, "silva_nr99_v138.1_train_set.fa.gz?download=1", multithread=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

taxa.print <- taxa 
rownames(taxa.print) <- NULL
head(taxa.print)
```


```{r}
knitr::opts_chunk$set(echo=FALSE, eval=FALSE)

#test taxo 2
taxa2 <- assignTaxonomy(seqtab.nochim, "silva_nr99_v138.1_wSpecies_train_set.fa.gz?download=1", multithread=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=FALSE, eval=FALSE)

#test taxo 2
taxa2.print <- taxa2 
rownames(taxa2.print) <- NULL
head(taxa2.print)
```

```{r}
knitr::opts_chunk$set(echo=FALSE, eval=FALSE)

#test taxo 3 fonctionne pas pk?
taxa3 <- assignTaxonomy(seqtab.nochim, "silva_species_assignment_v138.1.fa.gz?download=1", multithread=TRUE)
```

```{r}
knitr::opts_chunk$set(echo=FALSE, eval=FALSE)

#test taxo 3
taxa3.print <- taxa3 
rownames(taxa3.print) <- NULL
head(taxa3.print)
```

#Test supp taxo
```{bash}
knitr::opts_chunk$set(echo=TRUE, eval=FALSE)

wget http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

dna <- DNAStringSet(getSequences(seqtab.nochim)) 
load("SILVA_SSU_r138_2019.RData") 
ids <- IdTaxa(dna, trainingSet, strand="top", processors=NULL, verbose=FALSE) 
ranks <- c("domain", "phylum", "class", "order", "family", "genus", "species") 

taxid <- t(sapply(ids, function(x) {
        m <- match(ranks, x$rank)
        taxa <- x$taxon[m]
        taxa[startsWith(taxa, "unclassified_")] <- NA
        taxa
}))
colnames(taxid) <- ranks; rownames(taxid) <- getSequences(seqtab.nochim)
print(taxid)
```


#Handoff to phyloseq
```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

theme_set(theme_bw())
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

samples.out <- rownames(seqtab.nochim)

subject <- sapply(strsplit(samples.out, "D"), `[`, 1)

gender <- substr(seqtab.nochim,1,1)
subject <- substr(seqtab.nochim,2,999)

day <- as.integer(sapply(strsplit(samples.out, "D"), `[`, 2))

samdf <- data.frame(Subject=subject, Gender=gender, Day=day)

samdf$When <- "Early"
samdf$When[samdf$Day>100] <- "Late"
rownames(samdf) <- samples.out


print(samdf)

```


```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(samdf), 
               tax_table(taxa))
ps <- prune_samples(sample_names(ps), ps) 
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
ps
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

plot_richness(ps, x="Day", measures=c("Shannon", "Simpson"), color="Where")
```


```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu))
ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray")
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

plot_ordination(ps.prop, ord.nmds.bray, color="Where", title="Bray NMDS")
```

```{r}
knitr::opts_chunk$set(echo=TRUE, eval=TRUE)

top20 <- names(sort(taxa_sums(ps), decreasing=TRUE))[1:20]
ps.top20 <- transform_sample_counts(ps, function(OTU) OTU/sum(OTU))
ps.top20 <- prune_taxa(top20, ps.top20)
plot_bar(ps.top20, x="Fond", fill="Family") + facet_wrap(~Where, scales="free_x")
```



